src/freenet/node/LocationManager.java

```diff
@@ -246,7 +249,21 @@ public void run() {
                     }
                 }
             }
-        }, PITCH_BLACK_MITIGATION_STARTUP_DELAY);
+        }, (int) (node.fastWeakRandom.nextFloat() * PITCH_BLACK_MITIGATION_STARTUP_DELAY));
+    }
+
+    /** @return the millis to wait until the next pitch black check: tomorrow and at least 12 hours in the future. */
+    private long getNextPitchBlackMitigationDelayMillisecondsTomorrow(LocalDateTime now) {
+        return Math.max(HOURS.toMillis(12), getMillisUntilRandomTimeTomorrow(now));
+    }
+
+    private long getMillisUntilRandomTimeTomorrow(LocalDateTime now) {
+        LocalDateTime tomorrowHourUTC = now
```

Comment:
Why call it hour? That makes it sound like other components will be discarded. What about something like `tomorrowTime`?

Reply:
`tomorrowTime` sounds much better — thank you!